### ISSUE TITLE
Head-manager: TypeError: Cannot read property 'content' of null

### ISSUE THREAD

**Issue by liesislukas** [zeit/next.js#8220]
<Head> seems to work as expected but I see this error in the console and it seems like coming from nextjs. 

1. if there is an error on my side, the error message should do a better job describing possible issues and hint what needs to be fixed
2. if error is on nextjs side it should be fixed

![image](https://user-images.githubusercontent.com/2733862/62375886-02289e80-b548-11e9-8d0e-088b890d1401.png)

tried to isolate the issue and find when it happens. I removed all the surrounding code from _app.js

```
class MyApp extends App {
  render() {
    return (
      <div>
        <InitialHead/>
        #wefklj hi 123
      </div>
    );
  }
}
```

and InitialHead works perfectly fine, it's code:

```
import React from 'react';
import Head from 'next/head';
// import {NODE_ENV} from '../config';

// import Layout from 'components/Layout';
class InitialHead extends React.Component {
  render() {
    return (
      <Head>
        <title key={'title'}>Foo</title>
      </Head>
    );
  }
}

export default InitialHead;
```

The title is updated to "Foo"

----
"next": "^9.0.3",
macos 10.14.6
Chrome Version 76.0.3809.87 (Official Build) (64-bit)

**Comment by liesislukas** [comment]
On page, I have an svg logo which has some inline styles and some styles included from the scss file. Both places defined the max-width/height of the image. 

None works until i add `<Head></Head>` to _document.js render 🤔

Also can't find any documentation on why at _document Head is included as:

`import Document, {Head,Main, NextScript} from 'next/document';`

and in another place as: 

`import Head from 'next/head';`

What's the difference, when use which?

**Comment by timneutkens** [comment]
Please follow the issue template and provide a full reproduction.

**Comment by liesislukas** [comment]
the issue is that I didn't add head to _document but _app and other places. It would be awesome if that error would provide a hint that it needs to be in _document only.

also, this definitely should be somewhere in the documentation:
Question:
```
import {Head} from 'next/document';
import Head from 'next/head';

when & which one should be used?
```

>Hi, the difference between the two is the places they are meant to be used. The Head export from next/document is only meant for usage in _document. The Head export from next/head is meant for usage elsewhere in your app

https://spectrum.chat/next-js/general/whats-the-difference-between-these-imports~548d0862-72a5-4980-ac58-55f4445bdee4

**Comment by barrymay** [comment]
To add/respond to @liesislukas , I looked into this a bit more at the code level, and the reason this is all happening (I missed it too!) is that the **base** `_document.js` that next.js creates internally creates a dependency `meta[name=next-head-count]` selector that head-manager is depends upon and fails on.

The doc https://nextjs.org/docs#custom-document has a very good example to build upon and has the line

**`All of <Html />, <Head />, <Main /> and <NextScript /> are required for page to be properly rendered.`**

Also in that example, there's the import
```
import Document, { Html, Head, Main, NextScript } from 'next/document';
```

TLDR - If you create a _document.jsx at all, ensure it's using the correct imported elements named above.

**Comment by chuyik** [comment]
This errors happens because I use 'next/head' in other places rather than _document or _app.

**Comment by devknoll** [comment]
Seems reasonable that this should just warn. I’ll put up a PR later today

**Comment by chuyik** [comment]
@devknoll 
After review I find that `meta[next-head-count]` could be under `<body />` rather than `<head />` if we use html conditional comments.

Here's the reprodution:

**_pages/index.js**
```
import Head from 'next/head'

function Index () {
  return (
    <div>
      <Head>
        <div
          dangerouslySetInnerHTML={{
            __html: `
    <!--[if lt IE 10]>
      <style>
        .ant-modal {
          top: 300px\\9;
        }
      </style>
    <![endif]-->
    `,
          }}
        />
      </Head>
      nihao HEllo
    </div>
  )
}

export default Index
```

**Screenshot**
![image](https://user-images.githubusercontent.com/6262943/62669984-2f4fd380-b9c4-11e9-98c2-ed21fafe8273.png)

**Comment by timneutkens** [comment]
It's actually not put under body, you're injecting invalid html (`<div>` into the document head, which then shows wrong in Chrome devtools.

**Comment by barrymay** [comment]
Just my 2 cents, but I think a better solution for all this is a warning if next finds that the _document.tsx doesn't have the expected React components in the template (vs using plain html).

Alternatively, maybe this could be an eslint plugin rule, if there's any thought to creating a plugin...

**Comment by derit** [comment]
i found same issue, when i put javascript code inside ```<head> </head>```  and the solution i put the code outside ```<head> </head>```  , error solved
![Screenshot from 2020-01-17 13-24-47](https://user-images.githubusercontent.com/5389364/72589086-cf733680-392c-11ea-9437-32265a476869.png)

**Comment by chungheepark** [comment]
I found this error at nextjs antd-mobile example.
```js
// examples/with-antd-mobile/pages/_app.js
<Head>
    <meta
        name="viewport"
        content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"
        />
    <style global jsx>{`
        html,
        body {
            font-family: 'Helvetica Neue', 'Hiragino Sans GB', Helvetica, 'Microsoft YaHei', Arial;
            margin: 0;
        }
    `}</style>
</Head>
```

**Comment by anasalpure** [comment]
head-manager.ts:52 Uncaught (in promise) TypeError: Cannot read property 'tagName' of null
    at updateElements (head-manager.ts:52)
    at head-manager.ts:100
    at Array.forEach (<anonymous>)
    at head-manager.ts:99


### --I solve this problem by remove this line
<style id="jss-server-side"></style>
in <Header>  => next/head

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
